### PR TITLE
Working infra, with R53 etc

### PR DESCRIPTION
**Pull request by benjaminsherwood**

### CODE REVIEW

File: acm.tf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  validation_method = "DNS"
+}
+
+resource "aws_route53_record" "concourse_web_dl" {
+  for_each = {
+  for dvo in aws_acm_certificate.concourse_web_dl.domain_validation_options : dvo.domain_name => {
+    name   = dvo.resource_record_name
+    record = dvo.resource_record_value
+    type   = dvo.resource_record_type
+  }
```
Comment (dan-hill2802) [inline acm.tf]: I think this block needs one indent to the right

File: acm.tf [hunk 2]
```diff
@@ -0,0 +1,26 @@
+resource "aws_acm_certificate" "concourse_web_dl" {
+  domain_name       = local.fqdn
+  validation_method = "DNS"
```
Comment (dan-hill2802) [inline acm.tf:3]: should add the following (not sure why I can't add code suggestions)
`tags = local.common_tags`

File: concourse_web.tf [hunk 3]
```diff
@@ -1,15 +1,17 @@
+data "template_file" "concourse_web_userdata" {
```
Comment (dan-hill2802) [inline concourse_web.tf:1]: template files are a bad idea, you were better off with the `templatefile` function
https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file
Underlying reason is that the `template_file`data source uses a different provider and potentially different TF execution to main module

File: dns.tf [hunk 4]
```diff
@@ -1,6 +1,13 @@
...
+}
 
+resource "aws_route53_record" "concourse_web_lb" {
+  allow_overwrite = true
+  name            = local.fqdn
+  records         = [aws_lb.concourse_lb.dns_name]
+  ttl             = 60
+  type            = "CNAME"
+  zone_id         = data.aws_route53_zone.hosted_zone.zone_id
+}
```
Comment (dan-hill2802) [inline dns.tf:13]: ahh c'mon, you must have know I'd say something about this

File: dns.tf [hunk 5]
```diff
@@ -1,6 +1,13 @@
-//TODO: Requires a R53 HZ with some kind of sub-domain config allowing from a domain we can use for this
-//resource "aws_acm_certificate" "concourse" {
-//  domain_name       = local.fqdn
-//  validation_method = "DNS"
-//}
+data "aws_route53_zone" "hosted_zone" {
```
Comment (dan-hill2802) [inline dns.tf:1]: name suggestion:
```suggestion
data "aws_route53_zone" "public" {
```

**Review by dan-hill2802 — COMMENTED** [review]
I guess the commented out blocks is due to it being rather WIP?

File: dns.tf (hunk 4 above)
Comment (benjaminsherwood) [reply dns.tf:13]: lol damn newlines

File: concourse_web.tf (hunk 3 above)
Comment (benjaminsherwood) [reply concourse_web.tf]: as discussed, was due to inline one being a pita. now fixed.

**Pull request merged**